pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -672,14 +683,19 @@ func translateHeaderMatch(name string, in *networking.StringMatch, node *model.P
 		// Golang has a slightly different regex grammar
 		out.HeaderMatchSpecifier = &route.HeaderMatcher_PrefixMatch{PrefixMatch: m.Prefix}
 	case *networking.StringMatch_Regex:
-		if !util.IsIstioVersionGE14(node) {
-			out.HeaderMatchSpecifier = &route.HeaderMatcher_RegexMatch{RegexMatch: m.Regex}
-		} else {
-			out.HeaderMatchSpecifier = &route.HeaderMatcher_SafeRegexMatch{
-				SafeRegexMatch: &matcher.RegexMatcher{
-					EngineType: regexEngine,
-					Regex:      m.Regex,
-				},
+		switch m.Regex {
+		case ".*", "^.*", ".*$", "^.*$":
```

Comment:
This feels like the kind of thing that might be a bad idea, but I can't think of any reasons why. But would like some other reviewers to think about ways this may break

Reply:
Yes, it is tricky and heuristic, but I thought that would be practically helpful. When expressing "any string" in regex, the answer must be ".*" in most of cases.

Reply:
question: are these valid in ecmascript regex or the new safe regex engine in envoy? coz we send it as is to envoy. If all of these indicate same presence in the new safe regex engine (they should be), then lets optimize it out for speed. One less regex to ship to envoy as well.

Reply:
As far as I know, yes. But if someone can double-check, it would be great!

Reply:
I've realized that there are already the similar logics like a `isCatchAllMatch` or `isCatchAllRoute`. Those functions commonly interprets `*` in regex as "any string", so I've changed this part in order to align with such functions.